pandas/tests/plotting/test_datetimelike.py

```diff
@@ -628,7 +629,10 @@ def test_gap_upsample(self):
 
         idxh = date_range(low.index[0], low.index[-1], freq="12h")
         s = Series(np.random.default_rng(2).standard_normal(len(idxh)), idxh)
-        s.plot(secondary_y=True)
+
+        msg = "Plotting with mixed-frequency series is deprecated"
+        with tm.assert_produces_warning(Pandas4Warning, match=msg):
+            s.plot(secondary_y=True)
```

Comment:
it isn't clear to me why the warning is emitted here since `ax` isn't passed.  if that keyword isn't the relevant factor in deciding if the statefulness is needed, then #62442 may not achieve its goals